lib/src/pure_http/http_interceptor.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/src/base_request.dart';
+import 'package:http/src/client.dart';
+import 'package:http/src/base_client.dart';
+import 'package:http/src/streamed_response.dart';
+import 'package:http_mock_adapter/src/Exceptions.dart';
+import './http_response.dart';
+
+@Deprecated("This adapter class is an experimental feature")
+class ClientAdapter extends BaseClient {
```

Comment:
I think all of the `http` package code should be removed from this PR due to serving totally different purpose.

Also, the `http` imports that you use are from local system and not from `pubspec.yaml` file. Please include `http` as global dependency if you plan to add it as a feature, or else building would fail on other developers' systems.